### PR TITLE
Add a comment, to specify what to check.

### PR DESCRIPTION
**Pull request by SigmaSam**
# This is the line we added to accomplish the friendship v2
Created a mutual friendship method.

It changes the status of the friendship and creates another row to create the mutual friendship once the request is accepted.

# We advanced more than we had to in the Friendship v1 Milestone and added the functionality that we were supposed to add in this milestone.

### CODE REVIEW

**Review by kubilaycaglayan — CHANGES_REQUESTED** [review]
Hello Team 👋
Status: Changes Required ♻️

Great job! As you are telling, it is working nicely. 🎉👍

Let's cover some details to make it better, together 😊

## [REQUIRED] Tests

- [x] **Add unit and integration tests.** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/tests.png)  from  [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

💡 Currently, we only have 3 tests.

## [OPTIONAL] Routes

- [x] In the routes file, there are non CRUD routes linking to the actions in the controllers that could be named as any CRUD operation, you can use this naming system and the default RESTful routes created by rails instead of linking different named actions. You can check the usage of member or collection routes or the creation of a new controller in the links. [Reference for explaining member routes](https://guides.rubyonrails.org/routing.html#adding-more-restful-actions). [Reference for suggesting a new controller](http://jeromedalbert.com/how-dhh-organizes-his-rails-controllers/)

## That's it, team! After making the changes you can request another review 👍😊

With all my respect 🙏
> Kubilay Caglayan | Technical Support Engineer

File: spec/controllers/comments_spec.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  let!(:user1) { User.create(email: 'heroku@heroku', name: 'heroku', gravatar_url: 'heroku', password: '123456') }
+  let!(:post1) { Post.create(content: 'Test post', user_id: user1.id) }
+
+  it 'checks if a comment can be saved' do
+    expect(comment1).to be_invalid
+  end
+
+  it 'checks if a comment can be saved' do
+    expect(Comment.create(post_id: post1.id, user_id: user1.id, content: 'Test comment')).to be_valid
+  end
```
Comment (voscarmv) [inline spec/controllers/comments_spec.rb]: The `it` names for these tests are exactly the same, even though they test for opposite things. Please assign proper names for each of these tests, such as `"Comment creation fails with wrong parameters"` and `"Comment creation succeeds with valid parameters,"`.

File: spec/controllers/likes_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  let!(:like1) { Like.create(post_id: nil, user_id: nil) }
+  let!(:like2) { Like.create(post_id: post1.id, user_id: user1.id) }
+
+  it 'checks if a like is valid' do
+    expect(like1).to be_invalid
+  end
+
+  it 'checks if a like is valid' do
+    expect(like2).to be_valid
+  end
```
Comment (voscarmv) [inline spec/controllers/likes_spec.rb]: Same as before. Please use proper names to distinguish the nature of the tests.

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  end
+
+  it 'checks if users are friends' do
+    expect(tomas.friend?(pedro)).to be false
+  end
+
+  it 'checks if users are friends' do
+    expect(pedro.friend?(tomas)).to be true
+  end
+end
```
Comment (voscarmv) [inline spec/models/user_spec.rb]: Same as before. Please use proper names to distinguish the nature of the tests.

**Review by voscarmv — CHANGES_REQUESTED** [review]
# Impressive work, team! 🖥️ 👀 ✨ 

- Great work applying the requested changes and fulfilling most of the required features.

## However, some changes are still required... 🧰 🔨 🔧 

- The naming convention for your test suite are too ambiguous. Please read my code comments for details on how to fix this.

### Submit these changes and request a new code review! 😄 👍 

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Review by tirthajyoti-ghosh — APPROVED** [review]
# Approved ✅

## You have done an incredible job making this project 👏🧨💪! Just perfect 💯! 

### You have met all the requirements of the project. Go ahead and merge it! Onto the next project 🚀 !

Happy coding 👨‍💻 

<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

**Pull request merged**